src/wallet/wallet2.cpp

```diff
@@ -1313,7 +1314,7 @@ bool wallet2::set_daemon(std::string daemon_address, boost::optional<epee::net_u
   }
 
   MINFO("setting daemon to " << get_daemon_address());
-  return m_http_client.set_server(get_daemon_address(), get_daemon_login(), std::move(ssl_options));
+  return m_http_client->set_server(get_daemon_address(), get_daemon_login(), std::move(ssl_options));
 }
```

Comment:
All this massive spam all over the file should be avoided. You can probably do so easily by making m_http_client a reference, initialized in the ctor.

Reply:
Making `m_http_client` a reference will require it to be initialized in the constructor.  Since `abstract_http_client` is abstract, the default value could be an instance of `http_simple_client`, but that cannot be initialized in emscripten, so I do not have a suitable default value in that environment and execution fails.

I have updated the PR to use a reference which compiles but does not execute in emscripten because initializing `http_simple_client` in that environment fails.

Please advise.  Thanks.

Reply:
Off the top of my head, you can add a simple object, like:

struct redir {
redir(http_client *c): C(c) {}
operator http_client &() { return *c; }
http_client *C;
};

Then replace http_client *m_http_client with redir m_http_client in wallet2.h.
That should work AFAICT. Add default ctor if needed, and catch NULL if someone uses it before set.

Reply:
Two thoughts:

Use a `std::unique_ptr` instead of a raw pointer to clearly designate ownership. Unless `std::shared_ptr` behavior was necessary (it doesn't appear so at a glance).

Adding `*` and `->` stuff isn't really that hard to read, and we're now adding code just to reduce total line changes ... just why?

Reply:
To avoid all the diff spam changing lines that don't actually change.

Reply:
Seems like poor choice of words on my part - I know why you suggested it. But creating a class to prevent simple line changes seems incorrect. Especially since it hides pointer semantics - the value can be `NULL` but this is hidden at every usage point with an "auto" dereference.

Reply:
Not sure if you checked as the current diff does not show them anymore, but it changed every single RPC call for example. It was all over the place in wallet2.cpp.

Reply:
I also think it should be based on what is most correct and natural over these line diffs.  I've updated the PR to work with `std::unique_ptr` until moneromooo-monero's suggestion can be implemented for comparison.